src/components/ShareStructure/EditShareStructure.vue

```diff
@@ -0,0 +1,490 @@
+
+      if (this.activeIndex !== -1) {
+        const name = this.shareStructure.name
+        this.shareStructure.name = name.substr(0, name.indexOf(' Shares'))
+      }
+    }
+  }
+
+  // Methods
+  private validateForm (): void {
+    if (this.formValid) {
+      const shareStructure: ShareClassIF = this.addShareStructure()
+      this.emitAddShareStructureEvent(shareStructure)
+      this.resetFormAndData(false)
+    }
+  }
+
+  private emitAddShareStructureEvent (shareStructure: ShareClassIF): void {
+    if (this.isClass) {
+      this.emitAddEditShareClassEvent(shareStructure)
+    } else if (this.isSeries) {
+      this.emitAddEditShareSeriesEvent(shareStructure)
+    }
+  }
+
+  private addShareStructure (): ShareClassIF {
+    let shareStructureToAdd: ShareClassIF = { ...this.shareStructure }
+    if (this.activeIndex === -1) {
+      shareStructureToAdd.id = this.nextId
+    }
+    shareStructureToAdd.name = `${shareStructureToAdd.name} Shares`
+    shareStructureToAdd.hasMaximumShares = !this.hasNoMaximumShares
+    shareStructureToAdd.hasParValue = !this.hasNoParValue
+
+    if (!shareStructureToAdd.hasRightsOrRestrictions && shareStructureToAdd.series) {
+      shareStructureToAdd.series.forEach(series => {
+        series.action = ActionTypes.REMOVED
+      })
+    }
+    return shareStructureToAdd
+  }
+
+  private removeShareStructure (): void {
+    if (this.isClass) {
+      this.emitRemoveShareClassEvent(this.activeIndex)
+    } else if (this.isSeries) {
+      this.emitRemoveShareSeriesEvent(this.activeIndex, this.parentIndex)
+    }
+  }
+
+  private resetFormAndData (emitEvent: boolean): void {
+    this.$refs.shareStructureForm.reset()
+    if (emitEvent) {
+      this.emitResetEvent()
+    }
+  }
+
+  private changeMaximumShareFlag (): void {
+    if (this.hasNoMaximumShares) {
+      this.shareStructure.maxNumberOfShares = null
+    }
+  }
+
+  private changeParValueFlag (): void {
+    if (this.hasNoParValue) {
+      this.shareStructure.currency = null
+      this.shareStructure.parValue = null
+    }
+  }
+
+  private confirmSeriesRemoval (): void {
+    if (this.hasSeriesShares && !this.shareStructure.hasRightsOrRestrictions) {
+      // open confirmation dialog and wait for response
+      this.$refs.confirm.open(
+        'Remove Share Series',
+        'A share series exists for this class. Removing the Special Rights or Restrictions for this class' +
+        ' will remove all associated Share Series',
+        {
+          width: '45rem',
+          persistent: true,
+          yes: 'Remove share series',
+          no: null,
+          cancel: 'Cancel'
+        }
+      ).then(() => {
+        // if we get here, Yes was clicked
+        this.shareStructure.hasRightsOrRestrictions = false
+      }).catch(() => {
+        // if we get here, Cancel was clicked
+        this.shareStructure.hasRightsOrRestrictions = true
+      })
+    }
+  }
+
+  // Getters
+  get isClass (): boolean {
```

Comment:
FWIW, I like having my getters near the top of the script since they are kind of like "local properties".

Here, while reviewing, I had to "find on page" to find this declaration as it wasn't where I was expecting.

Reply:
They have been relocated in next commit!